Public/Src/Cache/ContentStore/Hashing/Pool.cs

```diff
@@ -86,8 +88,11 @@ public virtual PoolHandle Get()
 
         private void Return(T item)
         {
-            _reset?.Invoke(item);
-            _queue.Enqueue(item);
+            if (Size < _maxReserveInstances)

```

Comment:
don't you still need to reset the item even if you decide not to add it to the pool?

Reply:
Fair point. Most of our usages just used the _reset as a mechanism to clean the state and make the object valid for reuse. If we aren't reusing it that is unnecessary work. But one might also want to use the _reset as a mechanism for running logic with other side effects. So I suppose it would be safer to still run it even if we aren't going to be holding a reference to the object. I'll update that.